### PR TITLE
Enable vim-lsp for Python

### PR DESCRIPTION
**Pull request by chunliu**
Add vim-lsp configuration for Python to the code. Also add asyncomplete as another option for completion. 

vim-lsp works for Python with this change.

### CODE REVIEW

File: layers/+checkers/syntax-checking/packages.vim [hunk 1]
```diff
@@ -1,6 +1,2 @@
-if g:spacevim.vim8 || g:spacevim.nvim
-  MP 'w0rp/ale',  { 'on': [] }
-  call timer_start(200, 'spacevim#defer#ale')
-else
-  MP 'vim-syntastic/syntastic',     { 'on': 'SyntasticCheck' }
-endif
+MP 'dense-analysis/ale'
```
Comment (liuchengxu) [inline layers/+checkers/syntax-checking/packages.vim:1]: Why remove `syntastic`?

File: layers/+checkers/syntax-checking/config.vim [hunk 2]
```diff
@@ -1,36 +1,15 @@
...
+nmap <Leader>ep <Plug>(ale_previous)
+nnoremap <Leader>ts :ALEToggle<CR>
+" }
 
-  let g:ale_sign_error = '•'
-  let g:ale_sign_warning = '•'
-  let g:ale_echo_msg_error_str = g:spacevim.gui ? 'Error' : '✹ Error'
-  let g:ale_echo_msg_warning_str = g:spacevim.gui ? 'Warning' : '⚠ Warning'
-
-  let g:ale_completion_delay = 500
```
Comment (liuchengxu) [inline layers/+checkers/syntax-checking/config.vim:15]: Why remove these ALE options?

File: layers/+checkers/syntax-checking/packages.vim (hunk 1 above)
Comment (chunliu) [reply layers/+checkers/syntax-checking/packages.vim]: Sorry, the last commit was for my own config. I've revert it back.

File: layers/+checkers/syntax-checking/config.vim (hunk 2 above)
Comment (chunliu) [reply layers/+checkers/syntax-checking/config.vim]: I've reverted it back.

**Comment by liuchengxu** [comment]
Merged, thanks.

**Pull request merged**